server/src/test/java/io/crate/planner/CorrelatedJoinPlannerTest.java

```diff
@@ -131,7 +131,7 @@ public void test_prune_removes_unused_outputs_from_correlated_join() throws Exce
         );
         CorrelatedJoin join = new CorrelatedJoin(collect, selectSymbol, null);
         assertThat(join.outputs()).containsExactly(a, b, selectSymbol);
-        LogicalPlan pruned = join.pruneOutputsExcept(List.of(selectSymbol));
+        LogicalPlan pruned = join.pruneOutputsExcept(List.of(b, selectSymbol));
```

Comment:
This makes logically sense in respect to the result. `b` should the present in the outputs to keep if it should be in the the final result.

Reply:
Can you please check that query from https://github.com/crate/crate/issues/15901 works?
```
WITH cte1
AS (
	SELECT 'def' AS field1
		,5 AS unused_field1
	FROM information_schema.tables limit 1
	)
	,cte2
AS (
	SELECT 'abc' AS field2
		,cte1.field1
		,(
			SELECT count(*)
			FROM (
				SELECT 2 AS id
					,'abc' AS field2
					,'def' AS field1
				) subquery2
			WHERE 'abc' = subquery2.field2
				AND cte1.field1 = subquery2.field1
			) AS sorting_field
	FROM cte1
	)
SELECT (
		SELECT cte2.field2
		FROM cte2
		WHERE cte2.field2 = subquery1.target_field2
		ORDER BY sorting_field LIMIT 1
		) AS field3
FROM (
	SELECT 'abc' AS target_field2
		,'def' AS field1
	) subquery1;
```

Reply:
Can we treat the issue separately?

Reply:
ah ok.

Reply:
```
ClassCastException[class java.lang.String cannot be cast to class java.lang.Long (java.lang.String and java.lang.Long are in module java.base of loader 'bootstrap')]
```

Reply:
Can we handle these issues separately?

Reply:
You mean in another PR?
Probably yes, but fix which introduced `test_prune_removes_unused_outputs_from_correlated_join` is not yet released so we should do the follow up before we do 5.7.2

Reply:
This query works now (again).

Reply:
Thanks for checking!